src/nvim/api/ui.c

```diff
@@ -724,6 +724,17 @@ static void remote_ui_event(UI *ui, char *name, Array args, bool *args_consumed)
   push_call(ui, name, my_args);
 }
 
+static void remote_ui_option_set(UI *ui, String name, Object value)
+{
+  if (strequal(name.data, "ttimeout") || strequal(name.data, "ttimeoutlen")) {
```

Comment:
This should not be done. If TUI can use the option, so should any UI (the TUI is just another UI). Let's not introduce weird special cases without motivation.